File: Array/Find_pair_with_given_sum_in_array/Naive_Approach.java
class NaiveApproach {
	// Naive method to find a pair in an array with given sum
	public static void findPair(int[] A, int sum) {
		// Consider each element except last element
		for (int i = 0; i < A.length - 1; i++) {
			// Start from i'th element till last element
			for (int j = i + 1; j < A.length; j++) {
				// If desired sum is found, print it and return
				if (A[i] + A[j] == sum) {
					System.out.println("Pair found at index " + i + " and " + j);
					return;
				}
			}
		}
		// No pair with given sum exists in the array
		System.out.println("Pair not found");
	}

	public static void main(String[] args) {
		int[] A = {8, 7, 2, 5, 3, 1};
		int sum = 0;

		findPair(A, sum);
	}
}

File: Array/Find_pair_with_given_sum_in_array/Naive_Approach.py
# Naive method to find a pair in a list with given sum

def find_pair(A, sum):

    # consider each element except last element
    for i in range(len(A) - 1):

        # start from i'th element till last element
        for j in range(i + 1, len(A)):

            # if desired sum is found, print it and return
            if A[i] + A[j] == sum:
                print("Pair found at index", i, "and", j)
                return 

    # No pair with given sum exists in the list
    print("Pair not found")


if __name__ == "__main__":

    A = [8, 7, 2, 5, 3, 1]
    sum = 0

    find_pair(A, sum)
